
package org.jboss.test.ws.exception;

import javax.jws.WebMethod;
import javax.jws.WebService;
import javax.xml.bind.annotation.XmlSeeAlso;
import javax.xml.ws.RequestWrapper;
import javax.xml.ws.ResponseWrapper;


/**
 * This class was generated by the JAX-WS RI.
 * JAX-WS RI 2.1.7-04/11/2011 03:11 PM(mockbuild)-
 * Generated source version: 2.1
 * 
 */
@WebService(name = "ExceptionEndpoint", targetNamespace = "http://exception.ws.test.jboss.org/")
@XmlSeeAlso({
    ObjectFactory.class
})
public interface ExceptionEndpoint {


    /**
     * 
     * @throws UserException_Exception
     */
    @WebMethod
    @RequestWrapper(localName = "throwApplicationException", targetNamespace = "http://exception.ws.test.jboss.org/", className = "org.jboss.test.ws.exception.ThrowApplicationException")
    @ResponseWrapper(localName = "throwApplicationExceptionResponse", targetNamespace = "http://exception.ws.test.jboss.org/", className = "org.jboss.test.ws.exception.ThrowApplicationExceptionResponse")
    public void throwApplicationException()
        throws UserException_Exception
    ;

    /**
     * 
     */
    @WebMethod
    @RequestWrapper(localName = "throwRuntimeException", targetNamespace = "http://exception.ws.test.jboss.org/", className = "org.jboss.test.ws.exception.ThrowRuntimeException")
    @ResponseWrapper(localName = "throwRuntimeExceptionResponse", targetNamespace = "http://exception.ws.test.jboss.org/", className = "org.jboss.test.ws.exception.ThrowRuntimeExceptionResponse")
    public void throwRuntimeException();

    /**
     * 
     */
    @WebMethod
    @RequestWrapper(localName = "throwSoapFaultException", targetNamespace = "http://exception.ws.test.jboss.org/", className = "org.jboss.test.ws.exception.ThrowSoapFaultException")
    @ResponseWrapper(localName = "throwSoapFaultExceptionResponse", targetNamespace = "http://exception.ws.test.jboss.org/", className = "org.jboss.test.ws.exception.ThrowSoapFaultExceptionResponse")
    public void throwSoapFaultException();

}
